### DISCUSSION TITLE
Implementing WASI host functions on Windows platform

### DISCUSSION THREAD

**Discussion by moksh-pathak** [WasmEdge/WasmEdge#1360] [Q&A]
Hi, I am Moksh Pathak, a sophomore doing bachelors in Computer Science from India.

I had mentioned in the comments of #1227 that I would like to work on implementing the signal functions in windows. I have opened up this discussion to report my progress on #1227 as well as seek guidance and advice to approach and implement the solution in the most prudent and judicious way.

I have made the changes in the `environ-win.cpp` in my fork and pushed it on my forks' [branch](https://github.com/moksh-pathak/WasmEdge/blob/windows-signals/lib/host/wasi/environ-win.cpp). I didn't make a draft PR as I feel that this still needs some work; I am missing out on some blocks.

The C standard defines only 6 signals under `signals.h`, and these work well on Windows. For other POSIX signals, we need to find alternatives to make them work on Windows.

I referred to this [guide](https://docs.microsoft.com/en-us/previous-versions/ms811896(v=msdn.10)#signals-and-signal-handling) I found on internet and wrote the equivalent functions which work on console applications. For those signals which don't have an alternative in Windows API, I have commented NA below them.

I would like to get some input and advice on what I am missing out on and where I could improve to make this work.

I would also like to know the procedure to test my written code😅

Thanks and regards,
Moksh Pathak

**Comment by alabulei1** [comment]
@hydai  Could you help check this out?
